src/odemis/acq/align/fastem.py

```diff
@@ -18,22 +20,254 @@
+            self.asm_config = get_config_asm(self._multibeam, self._descanner, self._detector)
+
+            # loop over calibrations in list (order in list is important!)
+            for calib in self.calibrations:
+                # TODO return a sub-future when implemented for calibrations
+                self.run_calibrations(calib)
+
+                # def _pass_future_progress(sub_f, start, end):
+                #     f.set_progress(start, end)
+
+                # TODO Connect the progress of the sub-future to the main future when sub-futures are implemented
+                # sf.add_update_callback(_pass_future_progress)
+                # sf.result()
+
+                # remove from set of calibrations when finished
+                self._calibrations_remaining.discard(calib)
+
+                # In case the calibrations were cancelled by a client, before the future returned,
+                # raise cancellation error.
+                if self._cancelled:
+                    raise CancelledError()
+
+                # Update the time left for the calibrations remaining
+                self._future.set_progress(end=time.time() + self.estimate_calibration_time())
+
+        except CancelledError:
+            logging.debug("Calibration was cancelled.")
+            raise
+        except Exception as ex:
+            logging.error("Calibration failed: %s", ex, exc_info=True)
+            raise
+        finally:
+            # Remove references to the calibrations once all calibrations are finished/cancelled.
+            self._calibrations_remaining.clear()
+            self._scanner.blanker.value = True  # always blank the beam to reduce beam damage on sample
+            configure_asm(self._multibeam, self._descanner, self._detector, self._dataflow, self.asm_config)
+            logging.debug("Calibrations finished.")
+
+    def run_calibrations(self, calibration):
+        """
+        Run a calibration.
+        Note: All calibrations can be run on bare scintillator.
+        """
+        if calibration == OPTICAL_AUTOFOCUS:
+            autofocus_multiprobe.run_autofocus(self._scanner, self._multibeam, self._descanner, self._detector,
+                                               self._dataflow, self._ccd, self._stage)
+
+        if calibration == SCAN_ROTATION_PREALIGN:
+            scan_rotation_pre_align.run_scan_rotation_pre_align(self._scanner, self._multibeam, self._descanner,
+                                                                self._detector, self._dataflow, self._ccd)
+
+        if calibration == SCAN_AMPLITUDE_PREALIGN:
+            self.asm_config["multibeam"]["scanOffset"], self.asm_config["multibeam"]["scanAmplitude"] = \
+                scan_amplitude_pre_align.run_scan_amplitude_pre_align(self._scanner, self._multibeam, self._descanner,
+                                                                      self._detector, self._dataflow, self._ccd)
```

Comment:
Please make it 3 lines, so that if an error happens, it's clearer what's the issue. Something like:
```
soff, samp = run_scan_amplitude_pre_align(...)
self.asm_config["multibeam"]["scanOffset"] = soff
self.asm_config["multibeam"]["scanAmplitude"] = samp
```